casper/src/test/scala/coop/rchain/casper/MultiParentCasperMergeSpec.scala

```diff
@@ -53,34 +61,161 @@ class MultiParentCasperMergeSpec extends FlatSpec with Matchers with Inspectors
   }
 
   it should "handle multi-parent blocks correctly when they operate on stdout" ignore effectTest {
-    def echoContract(no: Int) = s"""new stdout(`rho:io:stdout`) in { stdout!("Contract $no") }"""
-    val time                  = System.currentTimeMillis()
-    HashSetCasperTestNode.networkEff(genesis, networkSize = 2).use { nodes =>
-      val deploys = Vector(
-        ConstructDeploy.sourceDeploy(echoContract(1), timestamp = time + 1),
-        ConstructDeploy.sourceDeploy(echoContract(2), timestamp = time + 2)
-      )
-      for {
-        block0 <- nodes(0).addBlock(deploys(0))
-        block1 <- nodes(1).addBlock(deploys(1))
-        _      <- nodes(0).receive()
-        _      <- nodes(1).receive()
-        _      <- nodes(0).receive()
-        _      <- nodes(1).receive()
+    def echoContract(no: Int) =
+      Rho(s"""new stdout(`rho:io:stdout`) in { stdout!("Contract $no") }""")
+    merges(echoContract(1), echoContract(2), Rho("Nil"))
+  }
 
-        //multiparent block joining block0 and block1 since they do not conflict
-        multiparentBlock <- nodes(0).addBlock(deploys(1))
-        _                <- nodes(1).receive()
+  // TODO: Peek rows/column
+  // Note this skips pairs that lead to infinite loops
+  it should "handle multi-parent blocks correctly when they operate on volatile produce/consume pairs" in effectTest {
+    // Sends (linear sends)
+    val S0 = Rho("@0!(0)")
+    val S1 = Rho("@0!(1)")
+    // Repeats (persistent sends)
+    val R0 = Rho("@0!!(0)")
+    val R1 = Rho("@0!!(1)")
+    // For-s (linear receives)
+    val F_ = Rho("for (_ <- @0) { 0 }")
+    val F0 = Rho("for (@0 <- @0) { 0 }")
+    val F1 = Rho("for (@1 <- @0) { 0 }")
+    // Contracts (persistent receives)
+    val C_ = Rho("contract @0(id) = { 0 }")
+    val C0 = Rho("contract @0(@0) = { 0 }")
+    val C1 = Rho("contract @0(@1) = { 0 }")
+    //FIXME add missing cases for in-deploy COMM-s wherever there's a pair without X (a COMM)
+    //FIXME all `conflictsForNow` should eventually be replaced with `merges`
+    Map(
+      "!X !X"        -> conflictsForNow(S0, S0, Nil),
+      "!X !4"        -> conflictsForNow(S0, S0, F_),
+      "!X (!4)"      -> merges(S0, S1 | F_, Nil),
+      "!X !C"        -> conflictsForNow(S0, S1, C1),
```

Comment:
the document describing mergeability (TDDM?) comments here:
`If one produce got consumed and one didn't, they must be sending on different names`
this case sends data on the same name (`@0`)  so the result should be `!C !C`

is the assumption that this will be fixed somehow? I think that this isn't a possible state as `!X` will never happen with a `C` present on a common name. 

the next case I'm able to follow. there we introduce `C` on the right and it has to conflict.

Reply:
missed the pattern match in C1